internal/provider/incident_catalog_entry_resource.go

```diff
@@ -259,12 +280,24 @@ func (r *IncidentCatalogEntryResource) Update(ctx context.Context, req resource.
 		}
 	}
 
-	result, err := r.client.CatalogV2UpdateEntryWithResponse(ctx, data.ID.ValueString(), client.UpdateEntryRequestBody{
-		Name:            data.Name.ValueString(),
-		Rank:            rank,
-		ExternalId:      data.ExternalID.ValueStringPointer(),
-		Aliases:         &aliases,
-		AttributeValues: data.buildAttributeValues(),
+	var updateAttributes *[]string
+	if !data.ManagedAttributes.IsUnknown() && !data.ManagedAttributes.IsNull() {
+		var attributeIDs []string
+		diags := data.ManagedAttributes.ElementsAs(ctx, &attributeIDs, false)
```

Comment:
actually you have a function for this, no?

Reply:
`ElementsAs` returns an error if any of the values are unknown which is actually very convenient here - if we try to call this API and some values are unknown that would be very freaky and we should abort!